zanata-client-commands/src/main/java/org/zanata/client/commands/init/SourceConfigPrompt.java

```diff
@@ -191,4 +218,92 @@ public String getExcludes() {
     public Set<String> getDocNames() {
         return docNames;
     }
+
+    private SrcDocsFinder makeSrcDocFinder() {
+        String projectType = pushOptions.getProjectType();
+        if (projectType.equalsIgnoreCase(File.name())) {
+            return new RawSrcDocsFinder(pushOptions);
+        } else {
+            return new OtherSrcDocsFinder(pushOptions);
+        }
+    }
+
+    interface SrcDocsFinder {
```

Comment:
Ah, I see these are really just wrappers that call to the push/pull strategy, to there is not the reuse problem. We can probably structure the whole client code to make reuse of pieces of the push/pull algorithm easier, but not necessary to do that now.